src/prompt/primary.go

```diff
@@ -139,4 +119,15 @@ func (e *Engine) writePrimaryRPrompt() {
 	e.write(strings.Repeat(" ", space))
 	e.write(e.rprompt)
 	e.write(terminal.RestoreCursorPosition())
+
+	if e.cached {
```

Comment:
This will never be `true`, since an assignment is missing after the engine is restored from cache.

Reply:
Good catch, it was there but I missed it after another refactor.